torch/__init__.py

```diff
@@ -426,11 +426,41 @@ def use_deterministic_algorithms(d):
     Note that deterministic operations tend to have worse performance than
     non-deterministic operations.
 
+    .. note::
+
+        :func:`use_deterministic_algorithms` is not guaranteed to prevent
+        nondeterministic behavior that is caused by an operation performing
+        multiple writes of different data to a single memory location. The
+        order of writes to a single location is not guaranteed, which can cause
+        nondeterminism. This could occur when an output tensor contains internal
+        memory overlaps or if an indexed operation is given duplicate indices.
```

Comment:
I thought we were going to conservatively warn to prevent nondeterminism from duplicate indices?